lsp/server/src/diagnostic/DiagnosticProducer.ts

```diff
@@ -19,4 +26,9 @@ export interface DiagnosticProducer<T> {
         textDocument: TextDocument,
         astNode: T
     ): Promise<Diagnostic[]>;
+
+}
+
+export interface DiagnosticMetaData {
```

Comment:
Diagnostic sent from server to client could have a data?: any.   DiagnosticMetaData structure is to enforce the data type.